### PR TITLE
Add exporting properties to gephi

### PR DESCRIPTION
**Pull request by tomasonjo**
Added support for exporting all properties from neo4j to gephi minus the reserved gephi parameter.  When approved I can also create a separate PR for 3.1 branch

### CODE REVIEW

**Comment by jexp** [comment]
Could you please add a test for the new functionality?
It should be an optional list of properties to be transferred passed as a param, b/c real world graphs can have *a lot* of properties and pretty large ones too.

**Comment by tomasonjo** [comment]
Updated with optional export properties

File: src/main/java/apoc/gephi/Gephi.java [hunk 1]
```diff
@@ -42,41 +44,42 @@ public static double doubleValue(PropertyContainer pc, String prop, Number defau
...
+    @Description("apoc.gephi.add(url-or-key, workspace, data, weightproperty, ['exportproperty']) | streams passed in data to Gephi")
+    public Stream<ProgressInfo> add(@Name("urlOrKey") String keyOrUrl, @Name("workspace") String workspace, @Name("data") Object data,@Name(value = "weightproperty",defaultValue = "null") String weightproperty,@Name(value = "exportproperties",defaultValue = "[]") List<String> exportproperties) {
         if (workspace == null) workspace = "workspace0";
         String url = getGephiUrl(keyOrUrl)+"/"+Util.encodeUrlComponent(workspace)+"?operation=updateGraph";
         long start = System.currentTimeMillis();
         HashSet<Node> nodes = new HashSet<>(1000);
         HashSet<Relationship> rels = new HashSet<>(10000);
         if (Graphs.extract(data, nodes, rels)) {
-            String payload = toGephiStreaming(nodes, rels, weightproperty);
+            String payload = toGephiStreaming(nodes, rels, weightproperty, exportproperties);
```
Comment (jexp) [inline src/main/java/apoc/gephi/Gephi.java]:
```
List<String> propertyNames = new ArrayList<>(exportproperties);
propertyNames.removeAll(RESERVED);
toGephiStreaming(nodes, rels, weightproperty, propertyNames);
```

File: src/main/java/apoc/gephi/Gephi.java [hunk 2]
```diff
@@ -42,41 +44,42 @@ public static double doubleValue(PropertyContainer pc, String prop, Number defau
...
 
-    private Map<String, Object> data(PropertyContainer pc, Map<String, Map<String, Object>> colors, String weightproperty) {
+    private Map<String, Object> data(PropertyContainer pc, Map<String, Map<String, Object>> colors, String weightproperty, List<String> exportproperties) {
         if (pc instanceof Node) {
             Node n = (Node) pc;
             String labels = Util.labelString(n);
             Map<String, Object> attributes = map("label", caption(n), "TYPE", labels);
             attributes.putAll(positions());
             attributes.putAll(color(labels,colors));
+            if (!exportproperties.isEmpty()) attributes.putAll(properties(n,exportproperties));
```
Comment (jexp) [inline src/main/java/apoc/gephi/Gephi.java]: if (!exportproperties.length > 0) attributes.putAll(n.getProperties(exportproperties));

File: src/main/java/apoc/gephi/Gephi.java [hunk 3]
```diff
@@ -42,41 +44,42 @@ public static double doubleValue(PropertyContainer pc, String prop, Number defau
...
+            String payload = toGephiStreaming(nodes, rels, weightproperty, exportproperties);
             JsonUtil.loadJson(url,map("method","POST"), payload).count();
             return Stream.of(new ProgressInfo(url,"graph","gephi").update(nodes.size(),rels.size(),nodes.size()).done(start));
         }
         return Stream.empty();
     }
 
-    private String toGephiStreaming(Collection<Node> nodes, Collection<Relationship> rels,String weightproperty) {
-        return Stream.concat(toGraphStream(nodes, "an", weightproperty), toGraphStream(rels, "ae", weightproperty)).collect(Collectors.joining("\r\n"));
+    private String toGephiStreaming(Collection<Node> nodes, Collection<Relationship> rels,String weightproperty, List<String> exportproperties) {
```
Comment (jexp) [inline src/main/java/apoc/gephi/Gephi.java]: String[] exportproperties

**Comment by tomasonjo** [comment]
Updated, removed the properties functions completely

**Comment by jexp** [comment]
Code Looks good !!

Sorry I forgot to mention that we also have to update the docs  @tomasonjo 

in overview.adoc and gephi.adoc

**Comment by tomasonjo** [comment]
added an example

**Pull request merged**